model-optimizer/extensions/middle/MakeKaldiConstReshapable_test.py

```diff
@@ -0,0 +1,116 @@
+"""
+ Copyright (C) 2018-2021 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+import unittest
+
+import numpy as np
+
+from extensions.middle.MakeKaldiConstReshapable import MakeKaldiConstReshapable
+from mo.front.common.partial_infer.utils import int64_array
+from mo.utils.ir_engine.compare_graphs import compare_graphs
+from mo.utils.unittest.graph import build_graph, result, regular_op_with_shaped_data, connect
+
+nodes = {
+    **regular_op_with_shaped_data('placeholder_1', [1, 13], {'kind': 'op', 'op': 'Parameter', 'shape': [1, 13]}),
+    **regular_op_with_shaped_data('splice_1', [1, 13], {'kind': 'op', 'op': 'Splice',
+                                                        'context': np.array([-2, -1, 0, 1, 2])}),
+    **regular_op_with_shaped_data('placeholder_2', [1, 26], {'kind': 'op', 'op': None}),
+    **regular_op_with_shaped_data('memory_in', [1, 5], {'kind': 'op', 'op': 'ReadValue',
+                                                        'shape': int64_array([1, 5])}),
+    **regular_op_with_shaped_data('memory_out', [1, 5], {'kind': 'op', 'op': 'Assign', 'shape': int64_array([1, 5])}),
+    **result('result'),
+    **regular_op_with_shaped_data('crop_in', [1, 4], {'kind': 'op', 'op': 'Crop', 'axis': 1, 'offset': 1, 'dim': 4}),
+    **regular_op_with_shaped_data('crop_out', [1, 1], {'kind': 'op', 'op': 'Crop', 'axis': 1, 'offset': 0, 'dim': 1}),
+    **regular_op_with_shaped_data('equal', [1, 1], {'kind': 'op', 'op': 'Equal'}),
+    **regular_op_with_shaped_data('select', [1, 26], {'kind': 'op', 'op': 'Select'}),
+    **regular_op_with_shaped_data('const_0', [1, 1], {'kind': 'op', 'op': 'Const', 'shape': [1, 1],
+                                                      'value': [0], 'data_type': np.float32}),
+    **regular_op_with_shaped_data('const_1', [1, 1], {'kind': 'op', 'op': 'Const', 'shape': [1, 1],
+                                                      'value': [0], 'data_type': np.float32}),
+    **regular_op_with_shaped_data('concat', [1, 5], {'kind': 'op', 'op': 'Concat'}),
+    **regular_op_with_shaped_data('memory', [1, 26], {'kind': 'op', 'op': 'Assign'}),
+
+    **regular_op_with_shaped_data('shape', None,  {'kind': 'op', 'op': 'ShapeOf'}),
+    **regular_op_with_shaped_data('crop_batch', None, {'kind': 'op', 'op': 'Crop', 'offset': int64_array([0])}),
+    **regular_op_with_shaped_data('crop_batch_dim', None, {'kind': 'op', 'op': 'Const', 'shape': [1],
```

Comment:
nit: you can use specific const creators

Reply:
Have not found const creator with both shape and value )